src/django/api/views/v1/production_locations.py

```diff
@@ -16,19 +20,75 @@
 from api.serializers.v1.production_locations_serializer \
     import ProductionLocationsSerializer
 from api.views.v1.index_names import OpenSearchIndexNames
+from api.permissions import IsRegisteredAndConfirmed
+from api.moderation_event_actions.creation.moderation_event_creator \
+    import ModerationEventCreator
+from api.moderation_event_actions.creation.location_contribution \
+    .location_contribution import LocationContribution
+from api.moderation_event_actions.creation.dtos.create_moderation_event_dto \
+    import CreateModerationEventDTO
+from api.models.moderation_event import ModerationEvent
+from api.throttles import DataUploadThrottle
+from api.constants import (
+    APIV1CommonErrorMessages,
+    NON_FIELD_ERRORS_KEY,
+    APIV1LocationContributionErrorMessages
+)
+from api.exceptions import ServiceUnavailableException
 
 
 class ProductionLocations(ViewSet):
     swagger_schema = None
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self.opensearch_service = OpenSearchService()
-        self.opensearch_query_builder = ProductionLocationsQueryBuilder()
-        self.opensearch_query_director = OpenSearchQueryDirector(
-            self.opensearch_query_builder
+    @staticmethod
+    def init_opensearch() -> Tuple[OpenSearchService, OpenSearchQueryDirector]:
```

Comment:
Can you, pls, apply the same OpenSearch initialization for `src/django/api/views/v1/moderation_events.py` in sake of consistensy?

Reply:
As for me, this class could be split:

- `validate_user_permissions`: This is validation that should be done by overriding the get_permissions function inside the ViewSet and creating a special class by instantiating permissions.BasePermission. See examples in the api.permissions module.
- `validate_uuid`, `fetch_moderation_event_by_uuid`, `validate_moderation_status`: These are methods that would fit perfectly in a serializer class.
- `handle_processing_error`: I think this can simply be raised directly since it only contains two lines of code.

These are just my thoughts to consider or not. I wouldn’t want to fix it within this PR; it’s just my thoughts.
I thought that service is something third-party.

Reply:
If my changes align with what you were expecting, could we please resolve this thread?

Reply:
Yes, let's resolve it.